src/components/structures/TimelinePanel.tsx

```diff
@@ -1058,7 +1059,7 @@ class TimelinePanel extends React.Component<IProps, IState> {
             } else {
                 cli.setRoomReadMarkers(
                     roomId,
-                    this.state.readMarkerEventId ?? "",
+                    this.state.readMarkerEventId ?? joinRoom(cli, [], roomId),
```

Comment:
Yeah this should do the job! Can you also add some tests around this use case ? Thanks

Reply:
joinRoom returns a `Promise<unknown>` so are you sure about that?

Reply:
I misread here, I though it was returning the join room event. I'll update my previous message.

Reply:
I think we can use `this.state.readMarkerEventId ?? JoinedRoomEvent(cli)[0],` 
Is this event will help us to achieve the required result?